src/services/measure-handler.ts

```diff
@@ -488,76 +540,80 @@ export const validateMeasureLookupTable = async (
-  const currentDataset = await DatasetRepository.getById(dataset.id);
-  const headers: CSVHeader[] = [];
-  for (let i = 0; i < tableHeaders.length; i++) {
-    headers.push({
-      index: i,
-      name: tableHeaders[i],
-      source_type: FactTableColumnType.Unknown
-    });
+  const quack = await duckdb();
+  await linkToPostgres(quack, revision.id, false);
+  try {
+    const preview = await quack.all(
+      pgformat(
+        'SELECT DISTINCT %I FROM %I ORDER BY %I ASC LIMIT %L;',
+        measure.factTableColumn,
+        'fact_table',
+        measure.factTableColumn,
+        sampleSize
+      )
+    );
+
+    const tableHeaders = Object.keys(preview[0]);
+    const dataArray = preview.map((row) => Object.values(row));
+    const currentDataset = await DatasetRepository.getById(dataset.id);
+    const headers: CSVHeader[] = [];
+    for (let i = 0; i < tableHeaders.length; i++) {
+      headers.push({
+        index: i,
+        name: tableHeaders[i],
+        source_type: FactTableColumnType.Unknown
+      });
+    }
+    const pageInfo = {
+      total_records: preview.length,
+      start_record: 1,
+      end_record: dataArray.length
+    };
+    const pageSize = preview.length < sampleSize ? preview.length : sampleSize;
+    return viewGenerator(currentDataset, 1, pageInfo, pageSize, 1, headers, dataArray);
+  } catch (error) {
+    logger.error(error, `Something went wrong trying to generate the preview of the measure column`);
+    throw error;
+  } finally {
+    await quack.close();
   }
-  const pageInfo = {
-    total_records: preview.length,
-    start_record: 1,
-    end_record: dataArray.length
-  };
-  const pageSize = preview.length < sampleSize ? preview.length : sampleSize;
-  return viewGenerator(currentDataset, 1, pageInfo, pageSize, 1, headers, dataArray);
 }
 
-async function getMeasurePreviewWithExtractor(dataset: Dataset, measure: Measure, quack: Database, lang: string) {
+async function getMeasurePreviewWithExtractor(dataset: Dataset, measure: Measure, revision: Revision, lang: string) {
   logger.debug(`Generating lookup table preview for measure ${measure.id}`);
-  const measureColumn = dataset.factTable?.find((col) => col.columnType === FactTableColumnType.Measure);
-  if (!measureColumn) {
-    logger.error(`Something went wrong trying to find the measure column for dataset ${dataset.id}`);
-    return viewErrorGenerators(500, dataset.id, 'patch', 'errors.dataset.measure_not_found', {});
-  }
-  if (!measure.measureTable || measure.measureTable.length === 0) {
-    logger.error(`Something went wrong trying to find the measure table for dataset ${dataset.id}`);
-    return viewErrorGenerators(500, dataset.id, 'patch', 'errors.dataset.measure_not_found', {});
+  const quack = await duckdb();
+  try {
+    await linkToPostgres(quack, revision.id, false);
+    const query = `SELECT * EXCLUDE(language) FROM measure WHERE language = '${lang.toLowerCase()}' ORDER BY sort_order, reference LIMIT ${sampleSize};`;
```

Comment:
Does this need a `pgformat()`?

Reply:
lang comes from the request - that's a potentially unsafe param.

We should probably just run everything through the sanitiser by default, regardless of where the param comes from.